clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -139,8 +142,11 @@ class ApiClient private (conf: APIConfigurations) {
       })
 
   def keyToStorageNamespace(key: StorageNamespaceCacheKey): String = {
-    val repo = repositoriesApi.getRepository(key.repoName)
 
+    val getRepo = new dev.failsafe.function.CheckedSupplier[Repository]() {
+      def get(): Repository = repositoriesApi.getRepository(key.repoName)
+    }
+    val repo = retryWrapper.wrapWithRetry(getRepo)
```

Comment:
```suggestion
    val repo = retryWrapper.wrapWithRetry(() => repositoriesApi.getRepository(key.repoName))
```

Reply:
Unfortunately, this won't compile with the Spark 2.4.7 + scala  2.11.12 combo